### PR TITLE
MueLu: Fix spread lumping when using BuildNew() version of filtered matrix

### PR DESCRIPTION
**Pull request by rstumin**
Auto-PR for SHA 02fe812

### CODE REVIEW

**Comment by cgcgcg** [comment]
@trilinos/framework @william76 Any idea why the autotester doesn't want to start?

**Comment by jwillenbring** [comment]
@cgcgcg We had an issue with a node and an effort to keep that node idle inadvertently made it so the GCC 8.3 test could not run anywhere. This issue should be corrected. I applied the retest label.

**Comment by cgcgcg** [comment]
@jwillenbring Great, thanks!

File: packages/muelu/src/Misc/MueLu_FilteredAFactory_def.hpp [hunk 1]
```diff
@@ -573,7 +576,7 @@ namespace MueLu {
  	if (row >= (LO)numRows) continue;
 	A.getLocalRowView(row, indsA, valsA);
 	for(LO k=0; k<(LO)indsA.size(); k++) {
-	  if(indsA[k] < (LO)numRows) {
+	  if ( (indsA[k] < (LO)numRows) && (valsA[k] != 0.0) ) {
```
Comment (cgcgcg) [inline packages/muelu/src/Misc/MueLu_FilteredAFactory_def.hpp:579]: Instead of checking for actual zeros, should this have a threshold?

Comment (rstumin) [reply packages/muelu/src/Misc/MueLu_FilteredAFactory_def.hpp:579]: The hard zero is ok ... but I should do the scalar traits stuff ... that I forgot about. The zero tests whether this is a really a nonzero or not. It is actually somewhat arbitrary (like it changes the math ... but it is not really sure what is better numerically) and does not cause the code to bomb either way. On the other hand, I should be using the scalar traits swill ... and zero() ... and stuff like that. I'll make that change .. hopefully in a few minutes.

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**